js/src/client.ts

```diff
@@ -746,6 +758,13 @@ export class Client {
     return true;
   }
 
+  protected async _getSettings() {
+    if (!this.settings) {
+      this.settings = await this._get("/settings");
```

Comment:
Should we store a promise instead? This does not necessarily dedupe settings requests